sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -595,7 +597,9 @@ public Object convertTimestampWithTimezone(java.time.OffsetDateTime offsetDateTi
                     return fromString(offsetDateTime.toString());
                 case TypeTags.OBJECT_TYPE_TAG:
                 case TypeTags.RECORD_TYPE_TAG:
-                    if (type.getName().equalsIgnoreCase(Constants.SqlTypes.DATETIME)) {
```

Comment:
You can do the same here.

Reply:
Updated. Commit id is 1200e15732e9aa11e651ec00907a9b346ece7359